rust-lang/cargo
# Support for descriptions on third-party subcommands in `cargo --list`

### What does this PR try to resolve?

This PR implements my proposal in #10662 to enable `cargo --list` to show descriptions of third-party subcommands, not only the subcommands which are built into Cargo.

### How should we test and review this PR?

I have included an ELF note containing a description in the most recent version of the `cargo-expand` crate. You can confirm that `cargo --list` is able to print a description for that subcommand (on Linux only, to begin with).

```console
$ cargo install cargo-expand --force
$ gh pr checkout 10663
$ cargo run -- --list
```
